Feature/cart

iuesver opened the pull request:
장바구니(localStorage)에 담긴 목록 정보에 따른 컴포넌트 구현

iuesver committed:
test

iuesver committed:
Revert "test"

This reverts commit 197776eaaf6d87ba44e0118c192441b95da2d1e2.

iuesver committed:
장바구니 컴포넌트 추가
1. localStorage 정보에 따른 목록 추가
2. localStorage가 비어있을 시 empty 컴포넌트 추가
3. 총 가격과 결제 버튼 추가

iuesver committed:
누락되었던 총 가격란 업데이트

in-woong commented on src/components/Cart.tsx:
```diff
@@ -0,0 +1,89 @@
+    const [count, setCount] = useState(1);
+    const [total, setTotal] = useState(0)
+
+    useEffect(() => {
```
localStorage에 cartItem을 넣어서 브라우저에서 cartItem을 관리하는 것이 아니라, Redux를 이용해서, cartItem을 관리해야할 것 같아요,
 그리고 창이 닫히거나 하는 상황에서 localStorage에 cartItem을 저장해서 연속성을 유지하는 것이 더 나은 방향이 아닐까 합니다!

in-woong commented on src/components/Cart.tsx:
```diff
@@ -0,0 +1,89 @@
+            <div className="cartItems">
+                {cart.length === 0 && <CartEmpty />}
+                {cart.length !== 0 && cart.map((item : item, idx : number) => (
+                    <div className="cartItem" key={idx}>
```
cartItem들이 임의로 지워지고, 다시추가되고 할 수 있는 부분이라면 item.id가 key가 되는 것이 더 좋을 것 같아요!
img의 alt 값도 추가해주시면 좋을 것 같습니다.

oinochoe replied:
관련 게시글 링크 드립니다 [[Link]](https://velog.io/@vlfflq2004/React-Index%EB%A5%BC-key%EB%A1%9C-%EC%93%B0%EB%A9%B4-%EC%95%88%EB%90%98%EB%8A%94-%EC%9D%B4%EC%9C%A0)

iuesver committed:
핵심 : localStorage환경에서 redux환경으로 변경
1. store 추가
2. 장바구니 redux store에 따른 렌더링
3. react-redux, redexjs/toolkit, swr 추가

iuesver committed:
Cart.tsx와 store.ts merge시 ProductDescription.tsx와 App.tsx의 변경점 요약 (package.json에 react-redux추가)
